packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -643,60 +643,111 @@ export class SDKClient {
+      createFileId = fileId;
 
+      const createFileRecord = await fileCreateTxResponse.getRecord(this.clientMain);
       this.captureMetrics(
         SDKClient.transactionMode,
-        fileAppendTx.constructor.name,
+        fileCreateTx.constructor.name,
         Status.Success,
-        await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
-        0,
+        createFileRecord.transactionFee.toTinybars().toNumber(),
+        createFileRecord?.contractFunctionResult?.gasUsed,
         callerName,
         interactingEntity,
       );
-    }
 
-    // Ensure that the calldata file is not empty
-    if (fileId) {
-      const fileSize = await (await new FileInfoQuery().setFileId(fileId).execute(client)).size;
+      if (fileId && callData.length > this.fileAppendChunkSize) {
+        const fileAppendTx = await new FileAppendTransaction()
+          .setFileId(fileId)
+          .setContents(hexedCallData.substring(this.fileAppendChunkSize, hexedCallData.length))
+          .setChunkSize(this.fileAppendChunkSize)
+          .setMaxChunks(this.maxChunks);
+        await fileAppendTx.execute(client);
+
+        this.captureMetrics(
+          SDKClient.transactionMode,
+          fileAppendTx.constructor.name,
+          Status.Success,
+          await this.calculateFileAppendTxTotalTinybarsCost(fileAppendTx),
+          0,
+          callerName,
+          interactingEntity,
+        );
+      }
+
+      // Ensure that the calldata file is not empty
+      if (fileId) {
+        const fileSize = await (await new FileInfoQuery().setFileId(fileId).execute(client)).size;
+
+        if (callData.length > 0 && fileSize.isZero()) {
+          throw new SDKClientError({}, `${requestIdPrefix} Created file is empty. `);
+        }
+        this.logger.trace(`${requestIdPrefix} Created file with fileId: ${fileId} and file size ${fileSize}`);
+      }
+    } catch (error: any) {
+      const sdkClientError = new SDKClientError(error, error.message);
+      let transactionFee: number | Hbar = 0;
+
+      // if valid network error utilize transaction id
+      if (sdkClientError.isValidNetworkError()) {
+        try {
+          const transactionRecord = await new TransactionRecordQuery()
+            .setTransactionId(fileCreateTx.transactionId!)
+            .setNodeAccountIds(fileCreateTx.nodeAccountIds!)
+            .setValidateReceiptStatus(false)
+            .execute(this.clientMain);
+          transactionFee = transactionRecord.transactionFee;
+
+          this.captureMetrics(
+            SDKClient.transactionMode,
+            fileCreateTx.constructor.name,
+            sdkClientError.status,
+            transactionFee.toTinybars().toNumber(),
+            transactionRecord?.contractFunctionResult?.gasUsed,
+            callerName,
+            interactingEntity,
+          );
+
+          this.hbarLimiter.addExpense(transactionFee.toTinybars().toNumber(), currentDateNow);
```

Comment:
This only adds the file create transaction cost.
What about the fileAppend transaction(s)?

Reply:
Updated.